internal/tss/utils/operation_builder.go

```diff
@@ -20,6 +20,11 @@ func BuildOperations(txOpXDRs []string) ([]txnbuild.Operation, error) {
 		if err != nil {
 			return nil, fmt.Errorf("decoding Operation FromXDR")
 		}
+
+		if op.GetSourceAccount() == "" {
+			return nil, fmt.Errorf("all operations must have a source account explicitly set")
+		}
+
```

Comment:
Enforce the caller to set a source account on every incoming operation, since we don't want the operations source accounts to fallback to the channel account who's signing the transaction.